src/main/java/com/alipay/oceanbase/rpc/mutation/result/MutationResult.java

```diff
@@ -47,15 +51,28 @@ public ObTableOperationType getOperationType() {
      */
     public long getAffectedRows() {
         long affectedRows = 0;
-        switch (result.getPcode()) {
-            case Pcodes.OB_TABLE_API_EXECUTE:
-                affectedRows = ((ObTableOperationResult) result).getAffectedRows();
-                break;
-            case Pcodes.OB_TABLE_API_QUERY_AND_MUTATE:
-                affectedRows = ((ObTableQueryAndMutateResult) result).getAffectedRows();
-                break;
-            default:
-                throw new ObTableException("unknown result type: " + result.getPcode());
+        if (result instanceof ObTableSingleOpResult) {
```

Comment:
could you based on pcode to return affected rows directly

Reply:
Fixed